### ISSUE TITLE
shade_surf restrictions and bug?

### ISSUE THREAD

**Issue by brandy125** [gnudatalanguage/gdl#1438]
Any simple shade_surf generates the following error message and the plot looks wrong:
```
GDL> a=dist(10) & shade_surf,a,findgen(10),findgen(10)
% SHADE_SURF: due to plplot restrictions, shaded surface is not entirely visible. Please try with zvalue=0.5 or greater.
GDL> 
```

With  ```zvalue=0.5```  applied the plot gets done and looks correct but appears only black/white and not shaded as in IDL:
```
GDL> a=dist(10) & shade_surf,a,findgen(10),findgen(10),zvalue=0.5
```
When the y-array is not strictly increasing then it produces an error message and aborts the plot:

```
GDL> a=dist(10) & shade_surf,a,findgen(10),-findgen(10),zvalue=0.5

*** PLPLOT ERROR, ABORTING OPERATION ***
plsurf3dl: Y array must be strictly increasing, aborting operation
GDL> 
```

Are this issues known? Can they be repaired easily?

**Comment by GillesDuvert** [comment]
@brandy125  these issues have occured mainly after the large rewriting of plotting routines in gdl that permit now to have the full set of 3D plots capabilities.
They are related to what I call a deficiency in plplot that impose the 3D surface to be seen from above, in a restricted range of z-values, for reasons I cannot fathom (reading the plplot's book at the chapter "Three Dimensional Surface Plots" shows how convoluted their reasoning) . 
Similarly, there is a restriction on the Y array that must be increasing etc...

The obvious solution would be as always, to avoid using plplot's surface algorithms and use our own. This probably will occur as I despair about the contouring and surface algorithms of plplot.

In the meantime, you will get back the 'shading' using device,decom=0 (plplot does not know about truecolor, either!)

This is a real bug for gdl now that 3D is OK and I keep this issue open until solved.

**Comment by GillesDuvert** [comment]
solved in #1515 , this is the maximum one can do with plplot shading/surface functions.
